### PR TITLE
First Tscircuit lint PR

### PR DESCRIPTION
**Pull request by Abse2001**
In this PR, I focused on making the plugin work and tested it on the demo repo using `bun link,` and it worked. I only added the banned words and the two param rules as a start. In the next step, I will finish the third rule context passing and work on the CLI command circuit, and we will see what other ideas we come up with.

image

image

### CODE REVIEW

**Comment by seveibar** [comment]
Nice!!! Will dig into this shortly!!!

File: README.md [hunk 1]
```diff
@@ -1 +1,96 @@
...
+      },
+    },
+    rules: {
+      "@tscircuit/lint/banned-words": "error",
+      "@tscircuit/lint/two-params": "error",
+    },
+  },
+];
+```
+
```
Comment (seveibar) [inline README.md:77]: We should avoid configs or having end users install eslint, lets introduce a binary command “tscircuit-lint” that allows a project to just put “tscircuit-lint” as a “lint” script

**Review by seveibar — APPROVED** [review]
Nice!!

File: README.md (hunk 1 above)
Comment (Abse2001) [reply README.md:77]: I will look into it

**Pull request merged**